lib/lol_dba/sql_generator.rb

```diff
@@ -56,7 +56,11 @@ def generate_instead_of_executing(&block)
 
       def migrations(which)
         migrator = nil
-        if ::ActiveRecord::VERSION::MAJOR == 4
+        if Gem::Version.new(ActiveRecord::VERSION::STRING) >= Gem::Version.new('5.2.0.rc1')
```

Comment:
you need to specify the `rc1` part? I don't like the idea of adding a special case for a release candidate version.

Reply:
I couldn't get truthy value with `5.2.0`. Maybe something like `5.1.999` would work.

Reply:
I poked around `Gem::Version` a bit and learned that prereleases sort between real releases (newest to oldest):

1. 5.2.0
1. 5.2.0.rc2
1. 5.2.0.rc1
1. 5.1.6

We could just check for a major of 5 and a minor of 2 which would cover the release candidates and the real release when it comes out. However, that would break as soon as 5.3 comes out.

We could check for a version > 5.1.999. That covers the release candidates and all future versions. It's a completely made up version and would break on patch 1000 (very unlikely).

Checking for  >= 5.2.0.rc1 is future proof, a real version, and the first version that contains the breaking change. To me, this feels like the best option.